docs/REST API/rest-api.md

```diff
@@ -8723,6 +8723,73 @@ Gets the result of a script that was executed.
 
 > Note: `created_at` is the creation timestamp of the script execution request.
 
+
+### Batch-run script
+
+Run a script on multiple hosts.
+
+The script will be added to each host's list of upcoming activities.
+
+`POST /api/v1/fleet/scripts/run/batch`
+
+#### Parameters
+
+| Name            | Type    | In   | Description                                                                                    |
+| ----            | ------- | ---- | --------------------------------------------                                                   |
+| script_id       | integer | body | **Required**. The ID of the existing saved script to run. |
+| host_ids        | array   | body |  List of host IDs.  Required if `filters` not specified. Only one of `host_ids` or `filters` may be included in the request.   |                                            |
+| filters | object  | body | See [filters](#filters3). Required if `host_ids` not specified. Only one of `host_ids` or `filters` may be included in the request.   |
+
+
+##### Filters
+
+| Name                              | Type    | Description   |
+| ---------------------             | ------- | ----------------------------------------------------------------------------------- |
+| query                             | string  | Search query keywords. Searchable fields include `hostname`, `hardware_serial`, `uuid`, and `ipv4`. |
+| status                            | string  | Host status. Can either be `new`, `online`, `offline`, `mia` or `missing`. |
+| label_id                          | number  | ID of a label to filter by.  |
+| team_id                           | number  | ID of the team to filter by. | 
```

Comment:
@rachaelshaw @sgress454  since scripts are scoped per-team, having a `team_id` filter is redundant right?

Reply:
I'd be +1 on sending it anyway, to ensure that API users are clear that they need to filter to hosts on the same team as the script, but up to @rachaelshaw

Reply:
@jacobshandling I agree with @sgress454; `team_id` is the way to say "run this on all the hosts on the team" even if it's redundant

This makes me realize we should throw an error if the `team_id` doesn't match the script's team (rather than just run it on no hosts). How about:
> Error: Cannot filter by team_id &lt;id&gt; because the specified script belongs to another team.

(This would only happen via the API, since the UI couldn't be in that state.)

Reply:
Okay sounds good

Reply:
In that case should we add a similar check when running by `host_ids`?

Reply:
Yeah, should already be covered: https://github.com/fleetdm/fleet/issues/26715#issuecomment-2816304333